proposals/route-blinding.md

```diff
@@ -0,0 +1,308 @@
+# Route Blinding
+
+## Table of Contents
+
+* [Proposal](#proposal)
+  * [Introduction](#introduction)
+  * [Overview](#overview)
+  * [Notations](#notations)
+  * [Requirements](#requirements)
+  * [Encrypted data](#encrypted-data)
+  * [Creating a blinded route](#creating-a-blinded-route)
+  * [Sending to a blinded route](#sending-to-a-blinded-route)
+  * [Receiving from a blinded route](#receiving-from-a-blinded-route)
+  * [Sample flow](#sample-flow)
+  * [Unblinding channels via fee probing](#unblinding-channels-via-fee-probing)
+* [Tips and Tricks](#tips-and-tricks)
+  * [Recipient pays fees](#recipient-pays-fees)
+  * [Dummy hops](#dummy-hops)
+  * [Wallets and unannounced channels](#wallets-and-unannounced-channels)
+  * [Blinded trampoline route](#blinded-trampoline-route)
+* [FAQ](#faq)
+  * [Why not use rendezvous](#why-not-use-rendezvous)
+  * [Why not use HORNET](#why-not-use-hornet)
+
+## Proposal
+
+### Introduction
+
+Route blinding is a lightweight technique to provide recipient anonymity by blinding an arbitrary
+amount of hops at the end of an onion path. It's more flexible than rendezvous because it lets
+senders arbitrarily update amounts and lock times, and reuse a blinded route multiple times (which
+is useful when retrying a failed route or using multi-part payments).
+
+The downside compared to rendezvous is that senders have more leeway to probe by changing various
+variables, so the scheme needs to explicitly defend against probing attacks and may be less private.
+
+Some use-cases where route blinding is useful include:
+
+* Recipient anonymity when receiving payments
+* Using unannounced channels in invoices without revealing them
+* Forcing a payment to go through a specific set of intermediaries that can witness the payment
+* Providing anonymous reply paths for onion messages
+
+### Overview
+
+At a high level, route blinding works by having the recipient choose an _introduction point_ and a
+route to himself from that introduction point. The recipient then blinds each node and channel
+along that route with ECDH. The recipient includes the blinded route and a _hop-blinding secret_ in
+the invoice, which allows each node in the blinded route to incrementally unblind the payloads.
+
+This scheme requires all the nodes in the blinded route and the sender to activate support for the
+feature. It only becomes effective once a big enough share of the network supports it.
+
+### Notations
+
+* A node `N(i)`'s `node_id` is defined as: `P(i) = k(i) * G` (`k(i)` is the node's private key).
```

Comment:
as mentioned in the #759 I believe it complicates reading the text to add the additional notation `P(i)` instead of just saying `N(i)`is always identified by its `node_id`

I guess this is more a stylistic question but I can tell you that it decreased readability of the document for me.

Reply:
The `P(i)` notation is really required if we want to detail every equation and allow readers to check the maths.
What you proposed is conflating the concept of "node" with its "node_id" so that `N(i)` and `P(i)` are the same thing, right?
I was afraid it would be confusing (I wanted `N(i)` to refer to the node as an entity), but I can see why it makes the proposal harder to read. I'll try updating this and see if it's better.

Reply:
Done in https://github.com/lightning/bolts/pull/765/commits/e1c98896e06c21ad64bc8f4f9604d6b55a073c39